File: src/main/java/com/server/ordering/domain/dto/request/RestaurantDataWithLocationDto.java
package com.server.ordering.domain.dto.request;

import com.server.ordering.domain.FoodCategory;
import com.server.ordering.domain.RestaurantType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@AllArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class RestaurantDataWithLocationDto extends RestaurantDataDto {

    private double latitude;
    private double longitude;

    public RestaurantDataWithLocationDto(String restaurantName, String ownerName, String address, int tableCount,
                                         FoodCategory foodCategory, RestaurantType restaurantType, Integer admissionWaitingTime,
                                         Integer orderingWaitingTime, double latitude, double longitude) {
        super(restaurantName, ownerName, address, tableCount, foodCategory, restaurantType, admissionWaitingTime, orderingWaitingTime);
        this.latitude = latitude;
        this.longitude = longitude;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/request/RestaurantDataWithLocationDto.kt
package com.byeoru.ordering_server.domain.dto.request

import com.byeoru.ordering_server.domain.FoodCategory
import com.byeoru.ordering_server.domain.RestaurantType

data class RestaurantDataWithLocationDto(val latitude: Double,
                                         val longitude: Double,
                                         val restaurantName: String,
                                         val ownerName: String,
                                         val address: String,
                                         val tableCount: Int,
                                         val foodCategory: FoodCategory,
                                         val restaurantType: RestaurantType,
                                         val admissionWaitingTime: Int,
                                         val orderingWaitingTime: Int)